### PR TITLE
added shallow routing

### PR DESCRIPTION
**Pull request by SimplySonu**
Closes #324
The page reloads each time a user clicked on the edit button on the status card.
To prevent this I have used shallow routing. 
The URL will get updated to '?edit=true' and the page won't get replaced, only the state of the route is changed.

### CODE REVIEW

**Comment by YashJain24-chief** [comment]
Please add which issue it closes

**Review by YashJain24-chief — APPROVED** [review]

File: src/context/index.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const AppWrapperContext: FC<Props> = ({ children }) => {
     const { edit: editQuery } = query;
     const isEditMode = editQuery === 'true';
     const onEditRoute = () => {
-        router.replace('?edit=true'); //TODO handle query better https://github.com/Real-Dev-Squad/website-status/issues/299 
+        router.push({ query: "edit=true" }, undefined, { shallow: true }); //TODO handle query better https://github.com/Real-Dev-Squad/website-status/issues/299
```
Comment (ivinayakg) [inline src/context/index.tsx:20]: the code is not linted properly, run the `lint-fix` script and then push it

**Review by ivinayakg — CHANGES_REQUESTED** [review]

File: src/context/index.tsx (hunk 1 above)
Comment (SimplySonu) [reply src/context/index.tsx:20]: ok will do that.

Comment (SimplySonu) [reply src/context/index.tsx]: Hi @ivinayakg I have fixed the changes can u check it once.

Comment (ivinayakg) [reply src/context/index.tsx]: have you ran the `lint-fix` script?

Comment (SimplySonu) [reply src/context/index.tsx]: yes

**Comment by RitikJaiswal75** [comment]
Please add a video describing the problem and then the video of solved page

**Comment by SimplySonu** [comment]
@RitikJaiswal75 


https://user-images.githubusercontent.com/75324160/198278494-8e308b27-06e8-42ef-aa56-1fdfecf29b5f.mp4

**Comment by SimplySonu** [comment]
@YashJain24-chief @RitikJaiswal75 @ivinayakg 
Please review.

File: src/context/index.tsx [hunk 2]
```diff
@@ -40,4 +40,4 @@ export const useAppContext = () => {
     }
     return context || {}
 }
-export default AppWrapperContext
```
Comment (RitikJaiswal75) [inline src/context/index.tsx:43]: Nit: Why does this file show changes over here, as you have done no changes to this line.

**Review by RitikJaiswal75 — APPROVED** [review]

File: src/context/index.tsx (hunk 2 above)
Comment (SimplySonu) [reply src/context/index.tsx:43]: I am not sure why this happened, there is no changes in that line.

**Review by YashJain24-chief — DISMISSED** [review]
Looks good to me

**Review by Maheima — CHANGES_REQUESTED** [review]

File: src/context/index.tsx [hunk 3]
```diff
@@ -16,8 +16,8 @@ const AppWrapperContext: FC<Props> = ({ children }) => {
     const { query } = router;
     const { edit: editQuery } = query;
     const isEditMode = editQuery === 'true';
-    const onEditRoute = () => {
-        router.replace('?edit=true'); //TODO handle query better https://github.com/Real-Dev-Squad/website-status/issues/299 
+	const onEditRoute = () => {
```
[the remark this answers is not in the document]
Comment (SimplySonu) [reply src/context/index.tsx]: i have made the changes. please check now

**Review by shubham-y — DISMISSED** [review]

**Review by YashJain24-chief — DISMISSED** [review]

**Review by Maheima — DISMISSED** [review]
LGTM!

**Comment by Maheima** [comment]
Please write test cases

**Comment by SimplySonu** [comment]
> Please write test cases
@Maheima 
Actually I think we cannot add a test case for this. 
Only a query parms is been added in the url. 
Please give me your thoughts on this if I am wrong.

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**